### PR TITLE
Add support for `[CP]` build phases

### PR DESCRIPTION
**Pull request by DanToml**
closes https://github.com/CocoaPods/cocoapods-deintegrate/issues/26 and adds more fixtures.

Also explains why https://github.com/CocoaPods/CocoaPods/issues/5764#issuecomment-241194494 were left after running deintegrate.

### CODE REVIEW

File: spec/command/deintegrate_spec.rb [hunk 1]
```diff
@@ -1,40 +1,52 @@
 require File.expand_path('../../spec_helper', __FILE__)
 
-def fixture_project(named)
```
Comment (DanToml) [inline spec/command/deintegrate_spec.rb]: These changes are mostly just refactoring to clean up adding specs for more versions of CocoaPods.

File: lib/cocoapods/deintegrator.rb [hunk 2]
```diff
@@ -54,7 +54,7 @@ def deintegrate_pods_libraries(target)
 
     def deintegrate_shell_script_phase(target, phase_name)
       phases = target.shell_script_build_phases.select do |phase|
-        phase.name && phase.name =~ /\A(\u{1F4E6}\s)?#{Regexp.escape(phase_name)}\z/
+        phase.name && phase.name =~ /#{Regexp.escape(phase_name)}\z$/
```
Comment (DanToml) [inline lib/cocoapods/deintegrator.rb]: The actual change. This uses the same logic as CocoaPods integrator, and checks sans prefix.

**Comment by orta** [comment]
Great work

**Comment by neonichu** [comment]
🚢

**Pull request merged**